framework/include/userobject/NearestPointBase.h

```diff
@@ -79,51 +79,36 @@ class NearestPointBase : public ElementIntegralVariableUserObject
    * @param p The point.
    * @return The UserObject closest to p.
    */
-  UserObjectType * nearestUserObject(const Point & p) const;
+  std::shared_ptr<UserObjectType> nearestUserObject(const Point & p) const;
```

Comment:
It depends. If you know who the ultimate owner is, and you know the lifetime of that object. It's fine to use a unique pointer and just pass raw pointers back to people who need access to that resource. I think shared pointers are more useful when the chain of custody is unknown or changes. (i.e. in this example does the ownership of the nearest user object really pass to the other user?)